### PR TITLE
Fix: Fixed invalid "Network folder error" when opening shell folders

### LINKED ISSUE #11883: Bug: "Not a folder" Popup prevents me from opening "This PC" folder.
### Description

When attempting to click on "This PC", a popup showing "Network Folder Error - The network name cannot be found" appears
https://user-images.githubusercontent.com/65097814/228394227-77799909-24e8-4c35-ba26-da229adb1cf5.mp4

### Steps To Reproduce

Open files (2.4.58.0)
Click on This PC
Popup should appear.

### Requirements

Fix the popup so that it shows the drives properly.

### Files Version

2.4.58.0

### Windows Version

Windows 11 22H2 22621.1413

### Log file

[debug.log](https://github.com/files-community/Files/files/11095348/debug.log)

### PR DESCRIPTION
**Pull request by hishitetsu**
**Resolved / Related Issues**
- [X] Were these changes approved in an issue or discussion with the project maintainers? In order to prevent extra work, feature requests and changes to the codebase must be approved before the pull request will be reviewed. This prevents extra work for the contributors and maintainers.
   Closes #11883 

**Validation**
How did you test these changes?
- [X] Did you build the app and test your changes?
- [X] Did you check for accessibility? You can use Accessibility Insights for this.
- [ ] Did you remove any strings from the en-us resource file?
   - [ ] Did you search the solution to see if the string is still being used? 
- [ ] Did you implement any design changes to an existing feature?
   - [ ] Was this change approved?
- [X] Are there any other steps that were used to validate these changes?
   This PR also fixes the issue with quick access widgets containing shell folders taking a long time to load.

### CODE REVIEW

File: src/Files.App/Filesystem/Drives.cs [hunk 1]
```diff
@@ -91,7 +91,8 @@ public static async Task<StorageFolderWithPath> GetRootFromPathAsync(string devi
 					return new StorageFolderWithPath(matchingDrive, rootPath);
 				}
 			}
-			else if (devicePath.StartsWith(@"\\", StringComparison.Ordinal)) // Network share
+			else if (devicePath.StartsWith(@"\\", StringComparison.Ordinal) &&
+				!devicePath.StartsWith(@"\\SHELL\", StringComparison.Ordinal)) // Network share
```
Comment (yaira2) [inline src/Files.App/Filesystem/Drives.cs:95]:
```suggestion
			// Network share or Shell folder
			else if (devicePath.StartsWith(@"\\", StringComparison.Ordinal) &&
				!devicePath.StartsWith(@"\\SHELL\", StringComparison.Ordinal)) // Network share
```

**Review by yaira2 — APPROVED** [review]
LGTM

File: src/Files.App/Filesystem/Drives.cs (hunk 1 above)
Comment (hishitetsu) [reply src/Files.App/Filesystem/Drives.cs:95]: No. Network share EXCLUDING shell folders.

Comment (hishitetsu) [reply src/Files.App/Filesystem/Drives.cs]: In any way, corrected comment position.

**Pull request merged**